src/rustc-driver-interacting-with-the-ast.md

```diff
@@ -1,12 +1,16 @@
 # Example: Type checking through `rustc_interface`
 
-`rustc_interface` allows you to interact with Rust code at various stages of compilation.
+The [`rustc_interface`] allows you to interact with Rust code at various stages of compilation.
 
 ## Getting the type of an expression
 
-To get the type of an expression, use the `global_ctxt` to get a `TyCtxt`.
+To get the type of an expression, use the [`global_ctxt()`] query to [get] a [`TyCtxt`].
 The following was tested with <!-- date-check: jan 2024 --> `nightly-2024-01-19`:
 
 ```rust
 {{#include ../examples/rustc-driver-interacting-with-the-ast.rs}}
 ```
+[`rustc_interface`]: https://doc.rust-lang.org/nightly/nightly-rustc/rustc_interface/index.html
+[`global_ctxt()`]: https://doc.rust-lang.org/nightly/nightly-rustc/rustc_interface/queries/struct.Queries.html#method.global_ctxt
```

Comment:
I think we somewhere decided to avoid including parentheses in links
```suggestion
To get the type of an expression, use the [`global_ctxt`] query to [get] a [`TyCtxt`].
The following was tested with <!-- date-check: jan 2024 --> `nightly-2024-01-19`:

```rust
{{#include ../examples/rustc-driver-interacting-with-the-ast.rs}}
```
[`rustc_interface`]: https://doc.rust-lang.org/nightly/nightly-rustc/rustc_interface/index.html
[`global_ctxt`]: https://doc.rust-lang.org/nightly/nightly-rustc/rustc_interface/queries/struct.Queries.html#method.global_ctxt
```